ddl/ddl_worker.go

```diff
@@ -434,6 +434,11 @@ func (w *worker) deleteRange(ctx context.Context, job *model.Job) error {
 
 func jobNeedGC(job *model.Job) bool {
 	if !job.IsCancelled() {
+		if job.Warning != nil && dbterror.ErrCantDropFieldOrKey.Equal(job.Warning) {
```

Comment:
ErrCantDropFieldOrKey is strange here.

Reply:
Please take a look at the comment, it is not strange at all :)

https://github.com/pingcap/tidb/blob/ab27d4918a5a07d565ce3ef98761b6de6e90c9cc/util/dbterror/ddl_terror.go#L193-L194